fastify-mocks.d.ts

```diff
@@ -0,0 +1,44 @@
+// fastify-mocks.d.ts
+
+// These types were mocked due to a desire to not utilize fastify as an unlisted peer dependency.
+// https://github.com/fastify/otel/issues/32
```

Comment:
I generically don't see those as needed. We don't use them anywhere else in our organization.

Reply:
Just to clarify, these are already in this codebase. Do you want me to remove them in this PR and suggest downstream install fastify as a devDependency as well?

Reply:
This was added as part of #32 to avoid TS conflicts with unlisted dependencies.

Don't have a strong opinion while keeping it or not, it makes the code more complex as we are doing mimic of Fastify types and we most likely will need to update them on new major if something changes.

Happy to go either way.

Reply:
I think they should go away. If we need to play ball with Sentry, let's try adding the peerDependency and see how many problems we are going to get.

Reply:
@metcoder95 want me to do that in this PR? Or someone want to do it in a separate PR?